compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributeSharedMemoryCopy.cpp

```diff
@@ -112,16 +112,36 @@ populateTilingCopyToWorkgroupMemPatterns(RewritePatternSet &patterns,
           StringAttr::get(patterns.getContext(), getVectorizeMarker())));
 }
 
+// Returns the vector size to use for the given genericOp considering its
+// operand/result element types.
+static int getBaseVectorSize(linalg::GenericOp genericOp) {
+  assert(genericOp.getNumDpsInits() == 1);
+  unsigned resultBW =
+      llvm::cast<MemRefType>(genericOp.getDpsInitOperand(0)->get().getType())
+          .getElementTypeBitWidth();
+  // Check the operand element types. If we have some sub-byte types there, make
+  // sure we at least read a full byte for the sub-byte-element operands.
+  unsigned operandBW = std::numeric_limits<unsigned>::max();
+  for (OpOperand *operand : genericOp.getDpsInputOperands()) {
+    unsigned b = getElementTypeOrSelf(operand->get()).getIntOrFloatBitWidth();
+    operandBW = std::min(operandBW, b);
```

Comment:
Dont you want to use `std::max` here. If you vectorize for the highest BW, all sub-byte bitwidths should also be vectorized?

Reply:
Right now the tiling and vector size is calculated from the highest BW. We need to scale up that vector size in order to make sure we it accounts for the smaller operand BW. E.g., if the operand is `i4` and result is `i32`, vectorizing to `i32` would mean we will have 4 elements to reach 128 bits. Given this is elementwise op, it would mean 4xi4 for operands too. I'm trying to scale it up to have each thread instead handle 8 elements so each thread at least reads one full byte for the i4 operand. This makes CodeGen easier on the SPIR-V side.

Reply:
im not sure I fully follow the logic, but you probably have a better handle on this.